src/GraphQL/TypeApi.hs

```diff
@@ -245,40 +250,47 @@ instance forall typeName interfaces fields m.
   buildResolver mHandler selectionSet = do
     -- First we run the actual handler function itself in IO.
     handler <- mHandler
-    -- We're evaluating an Object so we're collecting (name, Value)
-    -- pairs from runFields and build a GValue.Map with them.
-    r <- forM selectionSet $ \selection -> runFields @m @fields handler selection
-    pure $ GValue.toValue (GValue.mapFromList r)
+    -- We're evaluating an Object so we're collecting ObjectFields from
```

Comment:
Sure, I meant that if we validate the query there would be no need to check for duplicate fields. Even if the Object defines duplicate fields (which is another interesting bug we should catch, see discussion below) then we'd never hit this condition.